nucypher/blockchain/eth/interfaces.py

```diff
@@ -351,7 +351,7 @@ def _attach_provider(self,
         if not provider_uri and not provider:
             raise self.NoProvider("No URI or provider instances supplied.")
 
-        if provider_uri and not provider:
+        if provider_uri and (provider is None or provider is NO_BLOCKCHAIN_CONNECTION):
```

Comment:
```suggestion
        if provider_uri and not provider:
```

This constants will evaluate Falsy as long as the bool value has been set first.  I suggest setting it at the top of this module, or in `__init__.py`

```
NO_BLOCKCHAIN_CONNECTION.bool_value(False)
```

Reply:
Got it, thanks
✔️